org.adempiere.base/src/org/compiere/util/EMail.java

```diff
@@ -280,8 +290,15 @@ public String send ()
 			{
 				props.put("mail.smtp.starttls.enable", "true");
 			}
-
-			session = Session.getInstance(props, m_auth);
+			if (isOAuth2 && m_auth != null) {

```

Comment:
My 2 cents here, I think the validation `&& m_auth != null` is worthless at this point, as based on the code above isOAuth2 will never be true if m_auth is null. I understand it is defensive programming, but I think it could be abstracted more by creating a _userName_ variable (after all m_auth.getPasswordAuthentication().getUserName() is called twice in a scope of 20 lines. 

in that way, the m_auth != null would appear just once 
String userName = m_auth != null ? m_auth.getPasswordAuthentication().getUserName() : "";

I think both methods that receive the userName handle the null/empty value properly. 

But above the improvement of code cleanliness. I got the doubt if, in this way OAuth does not work with the SMTP flag off at all, AFAIK Microsoft can work without SMTP, but I am not sure. 

But by extracting the userName assignation, the userName could come from different sources in the future, (EmailUser, RequestUsedr, etc...), instead of making it depend on EMailAuthenticator. 

Anyway, just a thought I had by reviewing the code ... If you don't find it worthy, it is not a blocker.

Reply:
Thanks @d-ruiz - I dropped that part of the if - I added it as defensive programming, but as you said is related to IDEMPIERE-4723 I cleaned that piece.

About the flag, the mechanism configured in AD_Client is all related to SMTP, there is no prevision in iDempiere to sent email with a different protocol, just SMTP.

The two flags there are for "SMTP Authentication" and "SMTP SSL/TLS" - the unchecked SMTP Authentication is when a server doesn't require a user/name to allow sending emails.  That's very rare I think, maybe in corporate servers that would be allowed (but is discouraged).  AFAIK none of the known mail provider allows that because it would be a big source of spam without control.

When no authentication is set (again, discouraged but possible), then no need to pass a username or password at all - OAuth or not OAuth.